packages/antd/src/templates/BaseInputTemplate/index.tsx

```diff
@@ -46,8 +47,10 @@ export default function BaseInputTemplate<
 
   const handleNumberChange = (nextValue: number | null) => onChange(nextValue);
 
-  const handleTextChange = ({ target }: React.ChangeEvent<HTMLInputElement>) =>
-    onChange(target.value === "" ? options.emptyValue : target.value);
+  const handleTextChange = onChangeOverride
+    ? onChangeOverride
+    : ({ target }: React.ChangeEvent<HTMLInputElement>) =>
+        onChange(target.value === "" ? options.emptyValue : target.value);
```

Comment:
This default `onChange` behavior is duplicated across all themes. Might make more sense to refactor to utils, e.g.

```ts
// Is there a better name?
export default function useOnEventTargetStringChange<
  T = any,
  S extends StrictRJSFSchema = RJSFSchema,
  F extends FormContextType = any
>(
  options: WidgetProps<T, S, F>["options"],
  onChange: WidgetProps<T, S, F>["onChange"]
) {
  return useCallback(
    ({ target: { value } }: React.ChangeEvent<HTMLInputElement>) =>
      onChange(value === "" ? options.emptyValue : value),
    [onChange, options.emptyValue]
  );
}
```

I am leaning towards the idea that this logic should be extracted from `BaseInputTemplate` altogether, since it doesn't work for all cases. Then we wouldn't need an `onChangeOverrides` prop. The downside is that we would have to pass this in almost every time we use `BaseInputTemplate`:


```tsx
export default function CustomWidget() {
  const { onChange: _onChange, value, options, registry } = props;
  const BaseInputTemplate = getTemplate<"BaseInputTemplate", T, S, F>(
    "BaseInputTemplate",
    registry,
    options
  );
  const onChange = useOnEventTargetStringChange(options, _onChange)
  return (
    <BaseInputTemplate
      {...props}
      onChange={onChange}
    />
  );
}
```

I'm not sure if this verbosity + inversion is worth it for one (known) edge case...WDYT?

Reply:
A lower-lift option is to extend the default logic to also handle files (possibly being aware of the input type).

```ts
export default function useOnChangeEvent<
  T = any,
  S extends StrictRJSFSchema = RJSFSchema,
  F extends FormContextType = any
>(
  options: WidgetProps<T, S, F>["options"],
  onChange: WidgetProps<T, S, F>["onChange"],
  isFileInput?: boolean
) {
  return useCallback(
    (event: React.ChangeEvent<HTMLInputElement>) => {
      if (isFileInput) {
	// ...
      } else {
        onChange(value === "" ? options.emptyValue : value)
      }
    },
    [onChange, options.emptyValue, isFileInput]
  );
}
```